megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -2836,4 +2752,39 @@ protected Mounted getAntiAirAmmo(List<Mounted> ammoList, WeaponType weaponType,
 
         return returnAmmo;
     }
+
+    // Helper method that figures out the valid facing changes for the given shooter
+    private List<Integer> getValidFacingChanges(Entity shooter)
+    {
+    	// figure out all valid twists or turret turns
+        // mechs can turn:
+        //		one left, one right unless he has "no torso twist" quirk or is on the ground
+        //		two left, two right if he has "extended torso twist" quirk
+        // vehicles and turrets can turn any direction unless he has no turret
+        ArrayList<Integer> validFacingChanges = new ArrayList<Integer>();
+        if(shooter instanceof Mech &&
```

Comment:
You should replace these instanceof comparasons with equals comparisons to Entity.ETYPE_*.  It's a faster operation and the reason the ETYPEs exist.  So, you'd have (Entity.ETYPE_MECH == shooter.getEntityType()).  The only potential gotcha would be shooter being null, but you should have a null check that throws an NPE with a specific message at the start of the method as a best practice anyway.